src/cdk/datetime/datepicker.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  Optional,
+  ViewEncapsulation,
+  OnDestroy,
+  ContentChild,
+} from '@angular/core';
+import {Subject, Subscription} from 'rxjs';
+import {DateAdapter} from './date-adapter';
+import {CdkDatepickerInput} from './datepicker-input';
+import {CalendarView} from './calendar-view';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+
+/** Used to generate a unique ID for each datepicker instance. */
+let datepickerUid = 0;
+
+/** Component used to wire together the datepicker input and calendar view. */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-datepicker',
+  host: {
+    '[id]': 'id',
+  },
+  template: '<ng-content></ng-content>',
+  exportAs: 'cdkDatepicker',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class CdkDatepicker<D> implements OnDestroy {
+  /** The initial date of the datepicker. */
+  @Input()
+  get startAt(): D | null {
+    // If an explicit startAt is set we start there, otherwise we start at whatever the currently
+    // selected value is.
+    return this._startAt || (this._datepickerInput ? this._datepickerInput.value : null);
+  }
+  set startAt(value: D | null) {
+    this._startAt = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+  }
+  private _startAt: D | null;
+
+  /** Whether the datepicker should be disabled. */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled == null && this._datepickerInput ?
+        this._datepickerInput.disabled : !!this._disabled;
+  }
+  set disabled(value: boolean) {
+    const newValue = coerceBooleanProperty(value);
+
+    if (newValue !== this._disabled) {
+      this._disabled = newValue;
+      this._disabledChange.next(newValue);
+    }
+  }
+  private _disabled: boolean;
+
+  /** The calendar view displayed in this datepicker. */
+  @ContentChild(CalendarView) view: CalendarView<D>;
+
+  /** The unique id for the datepicker. */
+  private _uniqueId: string = `cdk-datepicker-${datepickerUid++}`;
```

Comment:
I don't think we need this as a separate variable, can just combine it with the `@Input() id` below